src/test/java/org/kizombadev/markdownparser/SyntaxAnalyzerTest.java

```diff
@@ -159,10 +160,76 @@ public void test() {
         assertThat(paragraph.getChildren().get(0).getChildren().get(0)).isTextElementWith("Foo");
-        assertThat(paragraph.getChildren().get(1)).isTextElementWith(" Bar");
+        assertThat(paragraph.getChildren().get(1)).isTextElementWith(" ");
+        assertThat(paragraph.getChildren().get(2)).isTextElementWith("Bar");
     }
+
+    @Test
+    public void testBlankBetweenBoldAndItalicBug() {
+        Syntax syntax = underTest.parse(ImmutableList.of(Token.DoubleStar, Token.createTextToken("Foo"), Token.DoubleStar, Token.Blank, Token.Star, Token.createTextToken("Bar"), Token.Star));
+        assertThat(syntax.getChildren()).hasSize(1);
+
+        Syntax paragraph = syntax.getChildren().get(0);
+        assertThat(paragraph).isSyntaxTypeOf(SyntaxType.PARAGRAPH);
+        assertThat(paragraph.getChildren()).hasSize(3);
+
+        assertThat(paragraph.getChildren().get(0)).isSyntaxTypeOf(SyntaxType.BOLD);
+        assertThat(paragraph.getChildren().get(0).getChildren().get(0)).isTextElementWith("Foo");
+
+        assertThat(paragraph.getChildren().get(1)).isTextElementWith(" ");
+
+        assertThat(paragraph.getChildren().get(2)).isSyntaxTypeOf(SyntaxType.ITALIC);
+        assertThat(paragraph.getChildren().get(2).getChildren().get(0)).isTextElementWith("Bar");
+    }
+
+    @Test
+    public void testBlankInUnorderedListBug() {
+        Syntax syntax = underTest.parse(ImmutableList.of(Token.Star, Token.Blank, Token.createTextToken("Foo"), Token.Blank, Token.NewLine, Token.Star, Token.Blank, Token.createTextToken("Bar")));
+        assertThat(syntax.getChildren()).hasSize(1);
+        assertThat(syntax.getChildren().get(0)).isSyntaxTypeOf(SyntaxType.UNORDERED_LIST);
+        assertThat(syntax.getChildren().get(0).getChildren().get(0)).isSyntaxTypeOf(SyntaxType.UNORDERED_LIST_ITEM);
+        assertThat(syntax.getChildren().get(0).getChildren().get(0).getChildren().get(0)).isTextElementWith("Foo");
+        assertThat(syntax.getChildren().get(0).getChildren().get(1)).isSyntaxTypeOf(SyntaxType.UNORDERED_LIST_ITEM);
+        assertThat(syntax.getChildren().get(0).getChildren().get(1).getChildren().get(0)).isTextElementWith("Bar");
+    }
+
+    @Test
+    public void testMutliBlanksBug() {
+        Syntax syntax = underTest.parse(ImmutableList.of(Token.createTextToken("Foo"), Token.Blank, Token.Blank, Token.createTextToken("Bar")));
+        assertThat(syntax.getChildren()).hasSize(1);
+        Syntax paragraph = syntax.getChildren().get(0);
+        assertThat(paragraph).isSyntaxTypeOf(SyntaxType.PARAGRAPH);
+        assertThat(paragraph.getChildren()).hasSize(3);
+        assertThat(paragraph.getChildren().get(0)).isTextElementWith("Foo");
+        assertThat(paragraph.getChildren().get(1)).isTextElementWith(" ");
+        assertThat(paragraph.getChildren().get(2)).isTextElementWith("Bar");
+
+    }
+
+    @Test
+    @Ignore
+    public void testBoldAndItalicBug() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Fix or remove this skipped unit test [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1607)